core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -46,9 +46,18 @@ private[spark] object SerDe {
     dis.readByte().toChar
   }
 
-  def readObject(dis: DataInputStream): Object = {
+  def readObject(dis: DataInputStream, typeName: String = ""): Object = {
```

Comment:
@shivaram , one problem is we don't have `val dataType = readObjectType(dis)` in `bytesToRow`.